.github/workflows/publish-package.yml

```diff
@@ -88,15 +88,41 @@ jobs:
       - name: Get sdist filename
         run: echo "SDIST_FILE=$(ls dist/*.tar.gz)" >> $GITHUB_ENV
 
-      - name: Build wheels
-        uses: pypa/cibuildwheel@v2.21.0
+      - name: Build wheels on MacOS and Linux
+        if: runner.os != 'Windows'
+        uses: pypa/cibuildwheel@v2.23.3
         with:
           package-dir: ${{ env.SDIST_FILE }}
           output-dir: dist/
         env:
           CIBW_TEST_COMMAND: >
               python -c "import ndindex; assert '1.' in ndindex.__version__"
-
+          CIBW_BEFORE_BUILD: >
+            case "$CIBW_BUILD_IDENTIFIER" in
+              *t)
+                echo "Installing pre-release Cython and NumPy for freethreading build $CIBW_BUILD_IDENTIFIER"
+                python -m pip install --pre --upgrade -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython numpy
+                ;;
+              *)
+                echo "Skipping pre-release dependencies for standard build $CIBW_BUILD_IDENTIFIER"
+                ;;
+            esac
+      - name: Build wheels on Windows
+        if: runner.os == 'Windows'
+        uses: pypa/cibuildwheel@v2.23.3
+        with:
+          package-dir: ${{ env.SDIST_FILE }}
+          output-dir: dist/
+        env:
+          CIBW_TEST_COMMAND: >
+              python -c "import ndindex; assert '1.' in ndindex.__version__"
+          CIBW_BEFORE_BUILD: >
+              python -c "import os, sys; sys.exit(0) if os.environ.get('CIBW_BUILD_IDENTIFIER','').endswith('t') else sys.exit(1)"
+              if ($LASTEXITCODE -eq 0) {
+                python -m pip install --pre --upgrade -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython numpy
+                if ($LASTEXITCODE -ne 0) { exit $LASTEXITCODE }
+              } else {
+              }
```

Comment:
This is wrong, will have to spin up a windows machine to check why. Any ideas are most welcome.

Reply:
I'd really try to avoid reinventing wheels like this, it's just plumbing. NumPy et all all use a bash script method that also works on Windows:
- https://github.com/numpy/numpy/blob/de784cd6ee53ffddf07e0b3f89bf22bda9fd92fb/pyproject.toml#L146
- https://github.com/numpy/numpy/blob/de784cd6ee53ffddf07e0b3f89bf22bda9fd92fb/tools/wheels/cibw_before_build.sh#L60